### DISCUSSION TITLE
How to submit all forms nested inside a <Form.provider> ?

### DISCUSSION THREAD

**Discussion by yoavTaieb** [ant-design/ant-design#42968] [Q&A] [answered]
How to submit nested form inside a <Form.provider>?

Example of how it would work:
`
<Form.Provider>
  <Form1>
  <Form2>
<button type="submit"> Submit </button>
</Form.provider>
`

The goal is to submit every child form on a button click and get the error in case one of them throws one

**Comment by Wxh16144** [comment]
As far as I know, there is no such feature. You can add refs to each form, and then do click events.

for example: https://codesandbox.io/s/ji-ben-shi-yong-antd-5-6-1-forked-6mn5d8?file=/index.tsx

**Reply by yoavTaieb** [reply]
Already tried this option, but my forms are dynamic and are created with a loop. Eg:
`<Form.Provider>  forms.map(<Form1>)  <button type="submit"> Submit </button> </Form.provider>
`
I don't see a way to integrate it with refs

**Reply by Wxh16144** [reply, marked as answer]
Store all refs in an array of refs? Like this: https://codesandbox.io/s/ji-ben-shi-yong-antd-5-6-1-forked-w7zkyn?file=/index.tsx
